system_tests/staker_test.go

```diff
@@ -462,8 +464,38 @@ func stakerTestImpl(t *testing.T, faultyStaker bool, honestStakerInactive bool)
 	if !stakerBWasStaked {
 		Fatal(t, "staker B was never staked")
 	}
+
+	if logHandler.WasLogged("data poster expected next transaction to have nonce \\d+ but was requested to post transaction with nonce \\d+") {
```

Comment:
is there a way to modify/add a new test to surface this error? (like when `createWalletFunc` is nil and staker's dataposter object is used to make a transaction)

Reply:
This issue is non deterministic, so it will be challenging to create a reliable test to surface this.

Also, a scenario in which createWalletFunc as nil while having an instance of a Staker's DataPoster doesn't exist in Nitro, so we would test an execution flow that will never occur anyway.